### PR TITLE
Doc for the new class based views

### LINKED ISSUE #1133: Add Documentation for the New Generic Class Based View Added in PR #1130
The PR merged in #1130 added new django style class based views.  These need to be added to the doc so users know they exist.

### PR DESCRIPTION
**Pull request by troygrosfield**
Added doc for the new class based views from #1130.

Fixes #1133.

### CODE REVIEW

**Comment by acdha** [comment]
Thanks! I'll review this later today

File: docs/views_and_forms.rst [hunk 1]
```diff
@@ -120,6 +125,41 @@ Views
...
+    # views.py
+    
+    from haystack.generic_views import SearchView
+    
+    class MySearchView(SearchView):
+        """My custom search view."""
+        
+        def get_context_data(self, *args, **kwargs):
+            context = super(MySearchView, self).get_context_data(*args, **kwargs)
+            # do something
```
Comment (acdha) [inline docs/views_and_forms.rst]: Do you think it'd be worth including an example here which would show how to e.g. filter the SearchQuerySet used? That's a really common task

Comment (troygrosfield) [inline docs/views_and_forms.rst]: Done.

**Comment by SalahAdDin** [comment]
When will i have the new release?

**Comment by dekoza** [comment]
Please mention that `page` should be renamed to `page_obj` in the template (if someone uses original template from haystack docs).

**Comment by troygrosfield** [comment]
@dekoza, isn't that implied by the fact that it's mentions it the new SearchView extends from Django's FormView?  A comment about changing "page" to "page_obj" sounds like something that should go into release notes or "upgrading to {next_version}" document.

**Comment by acdha** [comment]
@dekoza I would definitely second the idea of a brief “Upgrading” paragraph if you wanted to contribute something to the release notes or a note block for the new docs.

**Comment by troygrosfield** [comment]
@acdha, anything else we're looking for?

**Comment by dekoza** [comment]
OK, I'll add short "Upgrading" paragraph to what @troygrosfield added.

**Comment by dekoza** [comment]
@troygrosfield PR waiting :)

**Comment by troygrosfield** [comment]
@dekoza, I added your changes in manually.  See why here:

- https://github.com/troygrosfield/django-haystack/pull/2/files#issuecomment-72552929

**Comment by acdha** [comment]
https://github.com/django-haystack/django-haystack/commit/2ae56094ba1141b00c019c2acb5536f6d0adb3cf has the merged version with some extra links to the Django documentation.

**Comment by troygrosfield** [comment]
:+1:

**Comment by acdha** [comment]
You can view the build docs now and point out errors:

http://django-haystack.readthedocs.org/en/master/views_and_forms.html

I'm wondering whether we should point anyone unfamiliar with Django CBVs to @pydanny's http://twoscoopspress.org/products/two-scoops-of-django-1-6, http://ccbv.co.uk/, etc.

**Comment by troygrosfield** [comment]
Doc looks good to me.  

I think CBVs have been out long enough to where I don't think the CBV reference is needed IMHO.  I would keep the scope of the doc specific to django-haystack so it doesn't get bloated.  However, it's only a link and if someone felt strongly about putting it in there it's not the end of the world...

**Comment by acdha** [comment]
Yeah, I was wondering if I might just put that in the release notes

**Comment by SalahAdDin** [comment]
But, how can we do Faceted Search Views?

**Comment by acdha** [comment]
@SalahAdDin There's a FacetedSearchView class: https://github.com/django-haystack/django-haystack/blob/master/haystack/views.py#L153 That's a good point to clarify in the docs.

**Comment by SalahAdDin** [comment]
Yes, please, clarify in the docs, because if one person see the docs as it is, well, the person will be very confusing.

**Pull request closed without merging**